### PR TITLE
feat: support plain old node as strategy to get node version

### PR DESCRIPTION
**Pull request by matthewadams**
Support `node --version` as a strategy to get the node version.  Current code assumes `nvm` is in use.

## Description
Add function `node_native_version_prompt` to use `node --version` & refactor `node_version_prompt` to use new envvar `NODE_VERSION_STRATEGY` with values `nvm` (default) or `node` to call either `nvm_version_prompt` or `node_native_version_prompt`, respectively.

## Motivation and Context
Current code assumes `nvm` manages `node`.  I use `asdf`.

## How Has This Been Tested?
Tested locally.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [x] If my change requires a change to the documentation, I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] If I have added a new file, I also added it to ``clean_files.txt`` and formatted it using ``lint_clean_files.sh``.
- [ ] I have added tests to cover my changes, and all the new and existing tests pass.

### CODE REVIEW

**Comment by matthewadams** [comment]
FYI, existing tests pass, but there are no tests for powerline's `node` prompt text.  Just trying to get a quick & dirty done here.  Default behavior is backward compatible.

**Comment by cornfeedhobo** [comment]
When ever we touch stuff related to language interpreters, I like to bring up the need to support shims, which is how a lot of these version managers work now.

**Comment by matthewadams** [comment]
> When ever we touch stuff related to language interpreters, I like to bring up the need to support shims, which is how a lot of these version managers work now.

What does this mean with respect to this PR?

File: themes/base.theme.bash [hunk 1]
```diff
@@ -399,8 +402,21 @@ function nvm_version_prompt() {
 	fi
 }
 
+function node_native_version_prompt() {
+	local node
+	if which -s node; then
```
Comment (NoahGorny) [inline themes/base.theme.bash:407]: you can use `command_exists`, and probably should log here as well

**Review by NoahGorny — CHANGES_REQUESTED** [review]
Hi @matthewadams, thanks for your contribution!
I left a comment, but I am looking forward to merge this :)

File: themes/base.theme.bash (hunk 1 above)
Comment (matthewadams) [reply themes/base.theme.bash]: done :) -- feel free to massage as you see fit before merging

**Review by seefood — APPROVED** [review]
seems simple enough...

**Pull request merged**